### ISSUE TITLE
Asymmetries in hierarchical nodes

### ISSUE THREAD

**Issue by le-cds** [eclipse/elk#530]
In this example, the inner node might well be centred, but isn't:

![problem](https://user-images.githubusercontent.com/7465584/77670509-97b8e880-6f86-11ea-8bc8-84ad9df392fc.png)

**Comment by uruuru** [comment]
I just double-checked with the following elkt and would argue that it's the same issue as in #546.

![image](https://user-images.githubusercontent.com/7552459/79581590-ec094100-80ca-11ea-8dc9-b39f7012be44.png)

```
node outer {
	port p1
	port p2
	node inner {
		port p1 
		port p2	
	}
	edge p1 -> inner.p1
	edge inner.p2 -> p2
}  
```

**Comment by uruuru** [comment]
Also, there's significantly more space used when using `hierarchyHandling: INCLUDE_CHILDREN`: 

![image](https://user-images.githubusercontent.com/7552459/79903422-5398f680-8413-11ea-83de-a8e635ee9d3e.png)

**Comment by uruuru** [comment]
The issue is coupled with the width of the western hierarchical port dummy. 
![image](https://user-images.githubusercontent.com/7552459/79906989-23545680-8419-11ea-8003-5809f9153746.png)


A layer's size includes `margin.right` of the nodes. 
https://github.com/eclipse/elk/blob/44542874c97a175f82a6e7559b317ef6dbbcc86a/plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/LayerSizeAndGraphHeightCalculator.java#L70-L72

`margin.right` is set to a value larger than zero for hierarchical port dummies for this example in the `NodeMarginCalculator`. I spare the details as it's a bit intricate. 

The origin, however, is this line:
https://github.com/eclipse/elk/blob/44542874c97a175f82a6e7559b317ef6dbbcc86a/plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/LGraphUtil.java#L846

As a consequence, the x-coordinate of the *port on the hierarchical dummy node* ends up being `11`, which then results in margins being applied to the dummy node, which then results in a layer size larger than desired.

The achor change was recently introduced to fix #447. 

@le-cds your turn ;).

**Comment by le-cds** [comment]
Thanks for investigating this further! 🙂 

> It's working when using `hierarchyHandling: INCLUDE_CHILDREN`:
> 
> ![image](https://user-images.githubusercontent.com/7552459/79903524-85aa5880-8413-11ea-93ad-709f19232f81.png)

But here the node isn't centred either, is it?

**Comment by uruuru** [comment]
Yes, I presume it's the same issue. 

It would be good, if you double checked the layout view in your setup. At some point in time I didn't get the gap for the non-hierarchical layout either - which confused me.

**Comment by le-cds** [comment]
I'm currently working on #546 using this test graph, and I do get the gap there.

**Comment by le-cds** [comment]
Nice, I just spent over an hour tracking down the root cause of #546, just to find that you had already found **and** posted the exact problem here... 🤦 Anyway, #595 contains the fix for #530. I shall then look into the port label issue you describe above.

**Comment by le-cds** [comment]
Right, so this basic issue was a duplicate of #546 (or rather, the other way around, but well...).

> It's working when using hierarchyHandling: INCLUDE_CHILDREN:
> Unless one adds port labels ...

This is now tracked as #596.
